include/sparrow/array_data.hpp

```diff
@@ -0,0 +1,187 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <climits>
+#include <cstdint>
+#include <vector>
+
+#include "sparrow/buffer.hpp"
+#include "sparrow/dynamic_bitset.hpp"
+#include "sparrow/data_type.hpp"
+
+namespace sparrow
+{
+    struct array_data
+    {
+        data_descriptor type;
+        std::int64_t length = 0;
+        std::int64_t offset = 0;
+        using block_type = std::uint8_t;
+        using bitmap_type = dynamic_bitset<block_type>; 
+        // bitmap buffer and null_count
+        bitmap_type bitmap;
+        // Other buffers
+        std::vector<buffer<block_type>> buffers;
```

Comment:
Why define `bitmap_type` but not `buffer_type`? With or without aliases are both fine to me, although I would prefer if it was consistent.